app/models/advisory_lock.rb

```diff
@@ -0,0 +1,19 @@
+class AdvisoryLock
+  class << self
+    def allocate_sip_trunk_channel(sip_trunk, options = {}, &block)
+      with_lock("allocate_sip_trunk_channel:#{sip_trunk.id}", options, &block)
+    end
+
+    private
+
+    def with_lock(key, options, &block)
+      options.reverse_merge!(
+        timeout_seconds: 15.seconds,
+        transaction: true
+      )
+      ApplicationRecord.transaction do
```

Comment:
Any reason why we need a transaction here? I thought the transaction should be handled inside the passing block if any reason the caller needs it by themselves.